Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -1730,3 +1730,94 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Pull iTunes deltas from Collection
     C4Log("-------- %.3f sec with deltas, %.3f sec without; %.2fx speed",
           timeWithDelta, timeWithoutDelta, timeWithoutDelta/timeWithDelta);
 }
+
+// Test replication and conflict resolution behaviour when we send SG a rev history with a gap (number of new
+// revs between syncs > repl::tuning::kDefaultMaxHistory).
+// This test attempts to emulate two separate clients with the same doc, where one client's rev history lands
+// in the gap of the other client's rev history.
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Give SG a rev history with a gap", "[.SyncServerCollection]") {
```

Comment:
I think. it's more interesting to demonstrate the difference between the current set of constants, 50/50, v.s. ideal ones, infinity/infinity, or 200/200 in your example. I know we cannot do it without modifying the source code, but perhaps you can shrinking the size of numRev1/numRev2 proportionally to fit to maxTreeDepth.

Reply:
I don't quite understand your meaning. Would you mean potentially having two Catch Sections for the test, one with numRev values that won't cause a gap, one with numRev values that will cause a gap? Then we can show that the behaviour is the same with or without a gap?

Reply:
exactly